d3a_interface/validators/home_meter_validator.py

```diff
@@ -0,0 +1,159 @@
+"""
+Copyright 2018 Grid Singularity
+This file is part of D3A.
+
+This program is free software: you can redistribute it and/or modify it under the terms of the
+GNU General Public License as published by the Free Software Foundation, either version 3 of the
+License, or (at your option) any later version.
+
+This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See
+the GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License along with this program. If
+not, see <http://www.gnu.org/licenses/>.
+"""
+from typing import Optional
+
+from d3a_interface.constants_limits import ConstSettings
+from d3a_interface.exceptions import D3ADeviceException
+from d3a_interface.validators.base_validator import BaseValidator
+from d3a_interface.validators.utils import validate_range_limit
+
+GeneralSettings = ConstSettings.GeneralSettings
+HomeMeterSettings = ConstSettings.HomeMeterSettings
+
+
+class HomeMeterValidator(BaseValidator):
+    """Validator class for Home Meter devices."""
+
+    @classmethod
+    def validate(cls, **kwargs):
+        """Validate the rate values of the device."""
+        cls.validate_rate(**kwargs)
+
+    @classmethod
+    def validate_rate(cls, **kwargs):
+        """Validate rates of a Home Meter device."""
+        cls._validate_fit_to_limit(
+            fit_to_limit=kwargs.get("fit_to_limit"),
+            energy_rate_increase_per_update=kwargs.get("energy_rate_increase_per_update"),
+            energy_rate_decrease_per_update=kwargs.get("energy_rate_decrease_per_update"))
+        cls._validate_home_meter_consumption_rates(**kwargs)
+        cls._validate_home_meter_production_rates(**kwargs)
+
+    @staticmethod
+    def _validate_fit_to_limit(
+            fit_to_limit: Optional[bool], energy_rate_increase_per_update: Optional[bool],
+            energy_rate_decrease_per_update: Optional[bool]):
+        if fit_to_limit is True and (
+                energy_rate_decrease_per_update is not None
+                or energy_rate_increase_per_update is not None):
+            raise D3ADeviceException({
+                "misconfiguration": [
+                    "fit_to_limit and energy_rate_increase/decrease_per_update can't be set "
+                    "together."]})
+
+        if fit_to_limit is False and (
+                energy_rate_increase_per_update is None
+                or energy_rate_decrease_per_update is None):
+            raise D3ADeviceException(
+                {"misconfiguration": [
+                    "energy_rate_increase/decrease_per_update must be set if fit_to_limit is "
+                    "False."]})
+
```

Comment:
Hmm, this could be reused in the storage strategy, right?

Reply:
I think you're right! I can move it to the `utils` module for the moment and share it between the two validators. I have the feeling that these functions can become their own classes (e.g. `FitToLimitValidator`) but this might be for a different PR :)

Reply:
:-D Yes, leave the different class for now and only move this to a utility function. But why not making it a static function in the Baseclass? Or is this not a common practice?

Reply:
Ok, yes, convinced. It is really a util. Thanks a lot!